client/shaders/nodes_shader/opengl_fragment.glsl

```diff
@@ -52,6 +52,15 @@ vec4 applyToneMapping(vec4 color)
 }
 #endif
 
+#ifdef ENABLE_FAR_CONTRAST
+vec3 applyFarContrast(vec3 color)
+{
+	float cadist = vPosition.z / 5000.0;
+	color = color * (cadist * 9.0 + 1.0) - cadist * 1.8;
```

Comment:
what do you want to achieve here?
sub cadist * 1.8 from r, g and b ?
in such case you have to write it like that:
color = color * (cadist * 9.0 + 1.0) - vec3(cadist * 1.8);